packit_service/worker/handlers/testing_farm.py

```diff
@@ -135,11 +135,13 @@ def _get_or_create_group(
                 corresponding copr build.
 
         Returns:
-            The existing or created test run group.
+            The existing or created test run group and the test targets
+            to run the tests for.
 
         """
         if self._testing_farm_group_id is not None:
-            return TFTTestRunGroupModel.get_by_id(self._testing_farm_group_id)
+            group = TFTTestRunGroupModel.get_by_id(self._testing_farm_group_id)
```

Comment:
so what about the scenario when the tests are triggered after Copr build (=> for each target separately), and there is an exception for the last ran target, and the test handler is retried...in that case, here we get all the targets in the DB again..can this happen?

Reply:
I was thinking about this and thought that wouldn't occur but your scenario sounds like that this may be the case. It sounds like that to solve this, we need to save "target ID" not "group ID" in the handler during retries. From the code it looks like that the handler will be triggered for each failed test. Does this sound reasonable?

Reply:
maybe I am speculating too much, but if we would forward `target ID` when retrying, in case the tests were triggered by a comment (=> tests for all targets are submitted in one handler task) and e.g. we fail to submit tests for the first processed target, the handler retry task should still process all targets...wouldn't it make more sense to act depending on the configuration + targets overrides? :thinking:

Reply:
As discussed on chat, the task does not stop if one submission fails, therefore we can retry just the single target. Code adjusted, test case added.